src/main/java/org/folio/kafka/services/KafkaAdminClientService.java

```diff
@@ -0,0 +1,122 @@
+package org.folio.kafka.services;
+
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.kafka.admin.KafkaAdminClient;
+import io.vertx.kafka.admin.NewTopic;
+import org.apache.logging.log4j.Logger;
+import org.folio.kafka.KafkaConfig;
+import org.folio.kafka.exception.KafkaTopicsFileReadException;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static io.vertx.kafka.admin.KafkaAdminClient.create;
+import static org.apache.logging.log4j.LogManager.getLogger;
+import static org.folio.kafka.KafkaTopicNameHelper.formatTopicName;
+import static org.folio.kafka.services.KafkaEnvironmentProperties.getReplicationFactor;
+
+public class KafkaAdminClientService {
+
+  private static final Logger log = getLogger(KafkaAdminClientService.class);
+  private static final String KAFKA_TOPICS_FILE = "kafka-topics.json";
+  private final Supplier<KafkaAdminClient> clientFactory;
+
+  public KafkaAdminClientService(Vertx vertx) {
+    this.clientFactory = () -> create(vertx, KafkaConfig.builder()
+      .kafkaHost(KafkaEnvironmentProperties.getHost())
+      .kafkaPort(KafkaEnvironmentProperties.getPort())
+      .build().getProducerProps());
+  }
+
+  public Future<Void> createKafkaTopics(String environmentName, String moduleName, String tenantId) {
+    final List<NewTopic> topics = readTopics()
+      .map(topic -> topic.setName(formatTopicName(environmentName, moduleName, tenantId, topic.getName())))
+      .map(topic -> topic.setReplicationFactor(getReplicationFactor()))
+      .collect(Collectors.toList());
+    final KafkaAdminClient kafkaAdminClient = clientFactory.get();
+    return createKafkaTopics(1, topics, kafkaAdminClient)
+      .eventually(x -> kafkaAdminClient.close())
+      .onSuccess(result -> log.info("Topics created successfully"))
+      .onFailure(cause -> log.error("Unable to create topics", cause));
+  }
```

Comment:
What about the number of partitions for the topic? It should be specified while topic creation. Or it specified while deserialization?

Reply:
It was convetred from json file. But now we will use enum 
![image](https://user-images.githubusercontent.com/88006020/197757439-f4fc0c6b-2738-40bc-99cb-71e589e0e76e.png)